### PR TITLE
More than one execution even when success.

### PR DESCRIPTION
**Pull request by s2oBCN**
Exemple of what I saw: for a test method with two params with retry, if  first param fail (and then a retry is done), then the second param will be executed two times even if first time is SUCCES.

``` java
public class WebDriverRetryListenerTest {

    private int count = 0;

    @Test(dataProvider = "getData2")
    public void testParamsAndRFetry(int p1, String row) {
        System.out.println("testParamsAndRFetry -> count" + count++ + " , row:" + row);
        Assert.assertEquals("2on row", row);
    }

    @DataProvider
    public Object[][] getData2() {
        return new Object[][] { { 1, "1on row" }, { 2, "2on row" } };
    }
}
```

### CODE REVIEW

**Comment by juherr** [comment]
Could you add a test with the fix?

**Comment by s2oBCN** [comment]
I commited the test, but it will fail if Invoker is not changed. 

Thank you.

**Comment by juherr** [comment]
Your fix is breaking InvocationAndSuccessPercentageTest.
Could you have a look on it?

File: src/main/java/org/testng/internal/Invoker.java [hunk 1]
```diff
@@ -1067,7 +1067,7 @@ private ParameterBag createParameters(ITestNGMethod testMethod,
                                      new RegexpExpectedExceptionsHolder(m_annotationFinder, testMethod));
     final ITestClass testClass= testMethod.getTestClass();
     final List<ITestResult> result = Lists.newArrayList();
-    final FailureContext failure = new FailureContext();
+    FailureContext failure = new FailureContext();
```
Comment (juherr) [inline src/main/java/org/testng/internal/Invoker.java]: Why did you remove `final`?

File: src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java [hunk 2]
```diff
@@ -61,10 +63,14 @@ public void testAnnotationWithDataProvider(int paf, boolean test) {
       } else if (r2 == 0) {
         Assert.assertTrue(true);
       }
-    }
-    else if (paf == 0 || test == true) {
-      Assert.assertTrue(true);
-    }
+    } else if (paf == 0 && test == true) {
+			Assert.assertTrue(true);
```
Comment (juherr) [inline src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java]: Not a big deal, but could you follow existing indentation convention?

**Comment by juherr** [comment]
@s2oBCN Do you think you can update the pr with my 2 remarks? You can update CHANGES file too.

**Comment by s2oBCN** [comment]
Yes, I will do the update today.

**Comment by juherr** [comment]
:+1:

File: src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java [hunk 3]
```diff
@@ -61,9 +63,13 @@ public void testAnnotationWithDataProvider(int paf, boolean test) {
       } else if (r2 == 0) {
         Assert.assertTrue(true);
       }
-    }
-    else if (paf == 0 || test == true) {
-      Assert.assertTrue(true);
+    } else if (paf == 0 && test == true) {
```
Comment (cbeust) [inline src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java]: `test` is enough instead if `test == true`

File: src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java [hunk 4]
```diff
@@ -61,9 +63,13 @@ public void testAnnotationWithDataProvider(int paf, boolean test) {
       } else if (r2 == 0) {
         Assert.assertTrue(true);
       }
-    }
-    else if (paf == 0 || test == true) {
-      Assert.assertTrue(true);
+    } else if (paf == 0 && test == true) {
+	Assert.assertTrue(true);
```
Comment (cbeust) [inline src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java]: Unnecessary.

File: src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java [hunk 5]
```diff
@@ -61,9 +63,13 @@ public void testAnnotationWithDataProvider(int paf, boolean test) {
       } else if (r2 == 0) {
         Assert.assertTrue(true);
       }
-    }
-    else if (paf == 0 || test == true) {
-      Assert.assertTrue(true);
+    } else if (paf == 0 && test == true) {
+	Assert.assertTrue(true);
+	Assert.assertEquals(executionNumber, 3);
+    } else if (paf == 1 && test == true) {
```
Comment (cbeust) [inline src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java]: Ditto

**Comment by juherr** [comment]
@s2oBCN Could you rebase the PR and fix @cbeust remarks?

@rschmitt As you are the retry-analyzer top contributor, do you have any remark on this PR?

**Comment by s2oBCN** [comment]
Yes, I'll fix this in a week

**Comment by s2oBCN** [comment]
I'm sorry for the delay.
I see that there are some changes in the new RetryAnalyzerTest but the error remains unchecked, I'll make the changes to the new class InvocationCountTest befone this weekend.

**Comment by farazkhanQait** [comment]
Can you please tell me which version of testNg is having the fix as still i am having the same issue.

**Comment by juherr** [comment]
@farazkhanQait You can try the latest version: 7.0-beta3

**Comment by farazkhanQait** [comment]
thanks @juherr 
But whenever i am adding the dependency of 7.0-beta3 this is showing me error.
Missing artifact org.testng:testng:jar:7.0-beta3
will you please help me out

**Comment by farazkhanQait** [comment]
error
this is the error that i am getting when implementing beta verison

**Comment by juherr** [comment]
@farazkhanQait Use the beta version of the plugin. See http://testng.org/doc/download.html

**Comment by farazkhanQait** [comment]
Thanks @juherr 
But there is one issue with this failures are coming as skipped in it.
Have you guys have something for that too.
